quolance-api/src/main/java/com/quolance/quolance_api/services/impl/UserServiceImpl.java

```diff
@@ -98,18 +97,15 @@ public void resetPassword(UpdateUserPasswordRequestDto request) {
 
     @Override
     @Transactional
-    public UserResponseDto updateUser(UpdateUserRequestDto request) {
-        User user = SecurityUtil.getAuthenticatedUser();
-        user = userRepository.getReferenceById(user.getId());
+    public UserResponseDto updateUser(UpdateUserRequestDto request, User user) {
```

Comment:
This is a dangerous method and should be protected by admin role, we should keep the method intact as it is, and add this "new" implemmentation in the admin services, giving only the admin the ability to perform such thing

Reply:
Previously it was less dangerous because only the authenticated user could update itself (from the security Context), now we can feed it any user by this change which means user A can update user B.. i dont think this change is required